### PR TITLE
Improve fullscreen templates in landscape orientation

### PR DESCRIPTION
**Pull request by tonidero**
### Description
This modifies templates 1 and 5 so the top images don't use the full height of the screen when using devices in horizontal. This is not a problem when presenting as a dialog, since the width is limited, but if using the composables directly, it might be problematic.

### CODE REVIEW

**Comment by tonidero** [comment]
Screenshots change in: https://github.com/RevenueCat/purchases-android-snapshots/pull/3. Will hold this until the base PR is ready. snapshot tests will fail until that PR is merged.

File: ui/revenuecatui/src/main/kotlin/com/revenuecat/purchases/ui/revenuecatui/templates/Template1.kt
Comment (NachoSoto) [inline ui/revenuecatui/src/main/kotlin/com/revenuecat/purchases/ui/revenuecatui/templates/Template1.kt:145]: Oooh so nice

File: ui/revenuecatui/src/main/kotlin/com/revenuecat/purchases/ui/revenuecatui/templates/Template5.kt
Comment (NachoSoto) [inline ui/revenuecatui/src/main/kotlin/com/revenuecat/purchases/ui/revenuecatui/templates/Template5.kt:176]: What if the paywall isn't displayed full screen and is instead a dialog?

**Review by NachoSoto — COMMENTED** [review]
Awesome! Just one question.

File: ui/revenuecatui/src/main/kotlin/com/revenuecat/purchases/ui/revenuecatui/templates/Template5.kt
Comment (tonidero) [reply ui/revenuecatui/src/main/kotlin/com/revenuecat/purchases/ui/revenuecatui/templates/Template5.kt:176]: Hmm I was assuming this would pick the window size of the dialog, not the screen, but I was wrong, 
so this change would still apply there. As long as the width of the screen is bigger than the height, we will set the image as a percentage of the screen, instead of using an aspect ratio. This is not ideal, so trying to figure out a solution. Great catch! Thanks!

Comment (tonidero) [reply ui/revenuecatui/src/main/kotlin/com/revenuecat/purchases/ui/revenuecatui/templates/Template5.kt]: I changed the algorithm to use the template size instead of the window size, so we don't care if it's displayed as a dialog, as full screen, or anything the developer wants to do. This will cause a recomposition unfortunately... But I think it's the best solution

File: ui/revenuecatui/src/main/kotlin/com/revenuecat/purchases/ui/revenuecatui/templates/Template1.kt
Comment (vegaro) [inline ui/revenuecatui/src/main/kotlin/com/revenuecat/purchases/ui/revenuecatui/templates/Template1.kt:153]: Does this keep the aspect ratio? The `fillMaxWidth()` will fill the parent and the height will be reduced? So a portrait image will look squashed right?

File: ui/revenuecatui/src/main/kotlin/com/revenuecat/purchases/ui/revenuecatui/templates/Template5.kt
Comment (vegaro) [inline ui/revenuecatui/src/main/kotlin/com/revenuecat/purchases/ui/revenuecatui/templates/Template5.kt:189]: same question here about the aspect ratio

File: ui/revenuecatui/src/main/kotlin/com/revenuecat/purchases/ui/revenuecatui/templates/Template1.kt
Comment (tonidero) [reply ui/revenuecatui/src/main/kotlin/com/revenuecat/purchases/ui/revenuecatui/templates/Template1.kt:153]: No it won't. However, we are using `ContentScale.Crop` here, so the image won't look squashed but it will get cut, so depending on the image it might not look great.

I agree, this might not look great in some scenarios, but usually, this shouldn't be a problem as long as the width is not much larger than the height. So, for developers presenting as a dialog, this shouldn't be an issue I think (the dialog will not use the full available width in these scenarios.

Do you have any suggestions to handle this instead? This is the screenshots change after this: https://github.com/RevenueCat/purchases-android-snapshots/pull/3

Comment (vegaro) [reply ui/revenuecatui/src/main/kotlin/com/revenuecat/purchases/ui/revenuecatui/templates/Template1.kt:153]: I was missing the `ContentScale.Crop` 😄 

I was worried the image would look squashed, but it won't, so this should be fine

**Review by vegaro — APPROVED** [review]

File: ui/revenuecatui/src/main/kotlin/com/revenuecat/purchases/ui/revenuecatui/helpers/HelperFunctions.kt
Comment (NachoSoto) [inline ui/revenuecatui/src/main/kotlin/com/revenuecat/purchases/ui/revenuecatui/helpers/HelperFunctions.kt:21]: Nice: maybe extract `LocalConfiguration.current` in a `val` here to avoid calling it twice?

File: ui/revenuecatui/src/main/kotlin/com/revenuecat/purchases/ui/revenuecatui/PaywallDialog.kt
Comment (NachoSoto) [inline ui/revenuecatui/src/main/kotlin/com/revenuecat/purchases/ui/revenuecatui/PaywallDialog.kt:87]: > I changed the algorithm to use the template size instead of the window size, so we don't care if it's displayed as a dialog, as full screen, or anything the developer wants to do. This will cause a recomposition unfortunately... But I think it's the best solution

Isn't this still the window size?

Comment (tonidero) [reply ui/revenuecatui/src/main/kotlin/com/revenuecat/purchases/ui/revenuecatui/PaywallDialog.kt:87]: Yeah, but in this case, we can indeed use the window size I think, since it's what's going to determine whether we use a full height dialog or not, so in this case, we do know it's going to be displayed as a dialog, and we already have some logic so we don't display it as a full screen dialog for MEDIUM and EXPANDED devices.

Also note that this is just a refactor, but the logic hasn't changed in this PR.

Comment (NachoSoto) [reply ui/revenuecatui/src/main/kotlin/com/revenuecat/purchases/ui/revenuecatui/PaywallDialog.kt:87]: Oooh right right 👍🏻

File: ui/revenuecatui/src/main/kotlin/com/revenuecat/purchases/ui/revenuecatui/templates/Template1.kt
Comment (NachoSoto) [inline ui/revenuecatui/src/main/kotlin/com/revenuecat/purchases/ui/revenuecatui/templates/Template1.kt:141]: Maybe 
```suggestion
private fun HeaderImage(uri: Uri?, templateSize: IntSize) {
```

File: ui/revenuecatui/src/main/kotlin/com/revenuecat/purchases/ui/revenuecatui/templates/Template5.kt
Comment (NachoSoto) [inline ui/revenuecatui/src/main/kotlin/com/revenuecat/purchases/ui/revenuecatui/templates/Template5.kt:181]: Maybe we can add an `val IntSize.aspectRatio` extension?

**Review by NachoSoto — APPROVED** [review]

File: ui/revenuecatui/src/main/kotlin/com/revenuecat/purchases/ui/revenuecatui/extensions/IntSizeExtensions.kt
Comment (NachoSoto) [inline ui/revenuecatui/src/main/kotlin/com/revenuecat/purchases/ui/revenuecatui/extensions/IntSizeExtensions.kt:5]: `internal`?

**Pull request merged**
